experiment: Increase timeout for TestRunContainerWithRmFlagCannotStartContainer

vvoland opened the pull request:
I couldn't reproduce the flakiness on my Windows machine, but noticed that the run command itself takes a very long time to start. Maybe the removal in such case can take a long time too.
Increase timeout for this test just to test if it makes any difference.


<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

vvoland commented:
Yeah, the longer timeout didn't help. 120s should be enough I guess.

thaJeztah commented:
Yes, was also wondering if there's some deadlock or if it's another occurrence of Windows keeping a file locked ("file is in use by another process" kinda errors). 😞

thaJeztah commented:
😢 having a unique name was not the issue, looks like;

```
=== Failed
=== FAIL: github.com/docker/docker/integration-cli TestDockerCLIRunSuite/TestRunContainerWithRmFlagCannotStartContainer/sparkles_05b65481-6c5f-4291-a003-7aa1da193407_16 (24.60s)
    docker_cli_run_test.go:2777: timeout hit after 20s: waiting for container 'sparkles_05b65481-6c5f-4291-a003-7aa1da193407_16' to be removed
        --- FAIL: TestDockerCLIRunSuite/TestRunContainerWithRmFlagCannotStartContainer/sparkles_05b65481-6c5f-4291-a003-7aa1da193407_16 (24.60s)
```

thaJeztah commented:
looks like CI is happy (but not sure if you're still working on things beyond squashing etc)

vvoland committed:
libcontainer/windows: Fix process not being killed after stdio attach failure

Error check in defer block used wrong error variable which is always nil
if the flow reaches the defer. This caused the `newProcess.Kill` to be
never called if the subsequent attemp to attach to the stdio failed.
Although this only happens in Exec (as Start does overwrite the error),
this also adjusts the Start to also use the returned error to avoid this
kind of mistake in future changes.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
integration-cli: Enable TestExecStartFails on Windows

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
Common WorkingDirectory

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
Common ConsoleSize and EmulateConsole

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
Common CreateStdErrPipe

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
Common Environment

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
Common User

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
Inline setCommandLineAndArgs

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
Move User

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
libcontainerd/windows: Extract createProcessConfig

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
windows/Start: Get pid from newProcess

Do the same as Exec. Result is the same, just less indirection.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
libcontainerd/windows: Don't reap on failure

Synchronize the code to do the same thing as Exec.
reap doesn't need to be called before the start event was sent.
There's already a defer block which cleans up the process in case where
an error occurs.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
libcontainerd/windows: Common process creation

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
libcontainerd/windows: Extract createProcessWithStdio

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
enable tests

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
Revert "libcontainerd/windows: Extract createProcessWithStdio"

This reverts commit 68b036583b5556e5d0db1cfe6327e995892ecbf0.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
tesdt

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>
